core/scale/detail/fixed_witdh_integer.hpp

```diff
@@ -41,11 +43,11 @@ namespace kagome::scale::detail {
    * @param stream source stream
    * @return decoded value or error
    */
-  template <class T, typename I = std::decay_t<T>,
+  template <class T, class S, typename I = std::decay_t<T>,
             typename = std::enable_if_t<std::is_integral<I>::value>>
-  outcome::result<I> decodeInteger(common::ByteStream &stream) {
+  I decodeInteger(S &stream) {
     constexpr size_t size = sizeof(I);
-    static_assert(size == 1 || size == 2 || size == 4 || size == 8);
+    static_assert(size >= 1 && size <= 8);
```

Comment:
Is that possible that size can be less than 1 given an integral type?

Reply:
Probably no.